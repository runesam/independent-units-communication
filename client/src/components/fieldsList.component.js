import React from 'react';
import Grid from '@material-ui/core/Grid';
import propTypes from 'prop-types';

import {
    FormFieldComponent,
} from '.';

const FieldsListComponent = ({ fields, parent, ...rest }) => fields.map(field => (
    <Grid item xs={3} key={field.name} {...rest}>
        <FormFieldComponent
            field={field}
            parent={parent}
        />
    </Grid>
));

FieldsListComponent.propTypes = {
    fields: propTypes.arrayOf(propTypes.shape({})),
};

FieldsListComponent.defaultProps = {
    fields: [],
};

export default FieldsListComponent;
