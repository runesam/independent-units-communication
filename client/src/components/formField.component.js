import React from 'react';
import { Field } from 'redux-form';
import PropTypes from 'prop-types';

import { TextFieldComponent } from '.';

const FormFieldComponent = (props) => {
    const {
        field: {
            name,
            type,
            label,
            validators,
        },
    } = props;
    return (
        <Field
            name={name}
            type={type}
            label={label}
            placeholder={label}
            validate={validators}
            component={TextFieldComponent}
        />
    );
};

FormFieldComponent.propTypes = {
    field: PropTypes.shape({}).isRequired,
};

export default FormFieldComponent;
