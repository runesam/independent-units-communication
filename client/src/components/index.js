export { default as FieldsListComponent } from './fieldsList.component';
export { default as FormFieldComponent } from './formField.component';
export { default as TextFieldComponent } from './textField.component';
