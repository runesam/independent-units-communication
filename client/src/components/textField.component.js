import React from 'react';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import propTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

const styles = () => ({
    root: {
        minWidth: '100%',
        marginBottom: 7,
    },
});

const TextFieldComponent = (props) => {
    const {
        input,
        label,
        classes,
        meta: { touched, error },
        ...custom
    } = props;

    return (
        <div>
            <TextField
                label={label}
                className={classes.root}
                error={Boolean(touched && error)}
                InputLabelProps={custom.type === 'date' ? { shrink: true } : null}
                {...input}
                {...custom}
            />
            {Boolean(touched && error) && <Typography variant="subtitle2" color="error">{error}</Typography>}
        </div>
    );
};

TextFieldComponent.propTypes = {
	label: propTypes.string,
	meta: propTypes.shape({}),
	input: propTypes.shape({}),
	classes: propTypes.shape({}).isRequired,
};

TextFieldComponent.defaultProps = {
	meta: {},
    input: {},
	label: '',
};

export default withStyles(styles)(TextFieldComponent);
