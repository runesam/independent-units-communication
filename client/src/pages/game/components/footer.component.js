import React, { PureComponent } from 'react';
import propTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';
import Grid from '@material-ui/core/Grid';
import { withStyles } from '@material-ui/core';
import Typography from '@material-ui/core/Typography';

class Button extends PureComponent {
    static propTypes = {
        value: propTypes.number.isRequired,
        nextMove: propTypes.func.isRequired,
        disabled: propTypes.bool.isRequired,
    };

    onClick = () => {
        const { value, nextMove } = this.props;
        nextMove(value);
    };

    render() {
        const { value, disabled } = this.props;
        return (
            <Fab onClick={this.onClick} color="primary" size="large" disabled={disabled}>
                <Typography variant="h6">{value}</Typography>
            </Fab>
        );
    }
}

const Footer = (props) => {
    const {
        who,
        game,
        classes,
        nextMove,
    } = props;

    let disabled;

    if (who === 'ping') {
        disabled = game.length % 2 === 1;
    } else {
        disabled = game.length % 2 === 0;
    }

    return (
        <Grid
            container
            direction="row"
            alignItems="center"
            justify="space-around"
            className={classes.root}
        >
            {[-1, 0, 1].map(item => (
                <Button
                    key={item}
                    value={item}
                    disabled={disabled}
                    nextMove={nextMove}
                />
            ))}
        </Grid>
    );
};

Footer.propTypes = {
    who: propTypes.string.isRequired,
    nextMove: propTypes.func.isRequired,
    classes: propTypes.shape({}).isRequired,
    game: propTypes.arrayOf(propTypes.shape({})).isRequired,
};

const styles = theme => ({
    root: {
        bottom: 0,
        height: 80,
        position: 'fixed',
        backgroundColor: 'rgba(255, 255, 255, 0.7)',
        '& > button:disabled': {
            backgroundColor: theme.palette.primary.main,
            '& span h6': {
                color: '#9fdbf3',
            },
        },
    },
});

export default withStyles(styles)(Footer);
