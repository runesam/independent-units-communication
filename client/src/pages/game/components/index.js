import GameComponent from './game.component';
import HeaderComponent from './header.component';
import FooterComponent from './footer.component';
import ResultComponent from './result.component';
import PlayersMovesComponent from './playersMoves.component';

export {
    GameComponent,
    HeaderComponent,
    FooterComponent,
    ResultComponent,
    PlayersMovesComponent,
};
