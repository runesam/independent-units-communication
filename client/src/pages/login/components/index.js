export { default as LoginComponent } from './login.component';
export { default as LoginFormComponent } from './loginForm.component';
export { default as LoginFormWrapperComponent } from './loginFormWrapper.component';
