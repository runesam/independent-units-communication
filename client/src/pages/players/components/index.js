import TextFieldComponent from './textField.component';
import PlayerItemComponent from './playerItem.component';
import InvitationComponent from './invitation.component';
import PlayersListComponent from './playersList.component';
import InvitationFormComponent from './invitationForm.component';

export {
    TextFieldComponent,
    InvitationComponent,
    PlayerItemComponent,
    PlayersListComponent,
    InvitationFormComponent,
};
