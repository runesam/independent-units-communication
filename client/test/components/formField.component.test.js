import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Field } from 'redux-form';

import { FormFieldComponent } from 'components';

describe('<FormFieldComponent />', () => {
    const field = {
        name: 'username',
        label: 'Email',
        type: 'email',
    };

    it('matches the snapshot', () => {
        const wrapper = shallow(<FormFieldComponent parent="test" field={field} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('render the field with the right name', () => {
        const wrapper = shallow(<FormFieldComponent field={field} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find(Field)).toHaveLength(1);
        const { name } = wrapper.find(Field).props();
        expect(name).toEqual('username');
    });
});
