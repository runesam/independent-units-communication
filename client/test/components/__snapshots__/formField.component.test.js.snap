// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFieldComponent /> matches the snapshot 1`] = `
<Field
  component={[Function]}
  label="Email"
  name="username"
  placeholder="Email"
  type="email"
/>
`;

exports[`<FormFieldComponent /> render the field with the right name 1`] = `
<Field
  component={[Function]}
  label="Email"
  name="username"
  placeholder="Email"
  type="email"
/>
`;
